### DISCUSSION TITLE
Stability of `swc.bundle()`

### DISCUSSION THREAD

**Discussion by yklcs** [swc-project/swc#2207] [Q&A] [answered]
Hi, I was looking for a Node interface to Spack and found `swc.bundle()`. 

However, it doesn't seem to be documented currently, and type annotations ([`ModuleConfig`](https://github.com/swc-project/swc/blob/cbc823031042638c99fd58692ac94527d1ab1852/node-swc/src/spack.ts#L87)) seem to be incomplete.
Is `swc.bundle()` considered stable to be used in production currently? 

Thanks. 감사합니다!

**Comment by kdy1** [comment, marked as answer]
It's stable and actually, it is used by `deno bundle`.
But it's not documented well.

**Reply by yklcs** [reply]
@kdy1 Asking here because the thread is about `swc.bundle()`, and I already mentioned it in the OP:

`swc.bundle(options)` does not respect `options.module` , and [`ModuleConfig`](https://github.com/swc-project/swc/blob/cbc823031042638c99fd58692ac94527d1ab1852/node-swc/src/spack.ts#L87) is `{}`.
Is this intended behavior? #1217 was opened last year but went stale without triage.

Repro: https://gist.github.com/RocketLL/54e141b86c7ae9e71af0940cb1afc1fd

**Reply by kdy1** [reply]
It's not supported yet, and I'm not sure if it's usable from node js side.
I used it to bundle one private serverless projects, but I'm not sure about other usecases.
